src/golem-network/golem-network.ts

```diff
@@ -0,0 +1,379 @@
+import { DataTransferProtocol, DeploymentOptions, GolemDeploymentBuilder } from "../deployment";
+import { defaultLogger, Logger, YagnaApi } from "../shared/utils";
+import {
+  Demand,
+  DemandSpec,
+  DraftOfferProposalPool,
+  MarketApi,
+  MarketModule,
+  MarketModuleImpl,
+  MarketOptions,
+  OfferProposal,
+} from "../market";
+import { IPaymentApi, PaymentModule, PaymentModuleImpl, PaymentModuleOptions } from "../payment";
+import { ActivityModule, ActivityModuleImpl, IActivityApi, IFileServer } from "../activity";
+import { NetworkModule, NetworkModuleImpl } from "../network/network.module";
+import { EventEmitter } from "eventemitter3";
+import { LeaseProcess, LeaseProcessPool } from "../agreement";
+import { DebitNoteRepository, InvoiceRepository, MarketApiAdapter, PaymentApiAdapter } from "../shared/yagna";
+import { ActivityApiAdapter } from "../shared/yagna/adapters/activity-api-adapter";
+import { ActivityRepository } from "../shared/yagna/repository/activity-repository";
+import { AgreementRepository } from "../shared/yagna/repository/agreement-repository";
+import { IAgreementApi } from "../agreement/agreement";
+import { AgreementApiAdapter } from "../shared/yagna/adapters/agreement-api-adapter";
+import { ProposalRepository } from "../shared/yagna/repository/proposal-repository";
+import { CacheService } from "../shared/cache/CacheService";
+import { IProposalRepository } from "../market/offer-proposal";
+import { DemandRepository } from "../shared/yagna/repository/demand-repository";
+import { IDemandRepository } from "../market/demand";
+import { GftpServerAdapter } from "../shared/storage/GftpServerAdapter";
+import {
+  GftpStorageProvider,
+  NullStorageProvider,
+  StorageProvider,
+  WebSocketBrowserStorageProvider,
+} from "../shared/storage";
+
+export interface GolemNetworkOptions {
+  logger?: Logger;
+  api?: {
+    key?: string;
+    url?: string;
+  };
+  market?: Partial<MarketOptions>;
```

Comment:
this was created earlier, but as we are here it could be improved.
this seems to be unused anywhere, as is `deployment`... only `PaymentModule` has its options, which in addition seem to duplicate `PaymentSpec` in `DemandSpec`.... this needs to be clarified